src/IO/JLD2.jl

```diff
@@ -0,0 +1,77 @@
+"""
+    checkpointing_jld2(dst, stokes, thermal, particles, phases, time, igg)
+
+Save necessary data in `dst` as a jld2 file to restart the model from the state at `time`.
+If run in parallel, the file will be named after the corresponidng rank e.g. `checkpoint0000.jld2`
+and thus can be loaded by the processor while restarting the simulation.
+If you want to restart your simulation from the checkpoint you can use load() and specify the MPI rank
+by providing a dollar sign and the rank number.
+
+   # Example
+    ```julia
+    checkpointing_jld2(
+        "path/to/dst",
+        stokes,
+        thermal,
+        particles,
+        pPhases,
+        t,
+        igg,
+    )
+
+    ```
+"""
+checkpoint_name(dst) = "$dst/checkpoint.jld2"
+checkpoint_name(dst, igg::IGG) = "$dst/checkpoint" * lpad("$(igg.me)", 4, "0") * ".jld2"
+
+function checkpointing_jld2(dst, stokes, thermal, particles, phases, time)
+    fname = checkpoint_name(dst)
+    checkpointing_jld2(dst, stokes, thermal, particles, phases, time, fname)
+    return nothing
+end
+
+function checkpointing_jld2(dst, stokes, thermal, particles, phases, time, igg::IGG)
+    fname = checkpoint_name(dst, igg)
+    checkpointing_jld2(dst, stokes, thermal, particles, phases, time, fname)
+    return nothing
+end
+
+function checkpointing_jld2(dst, stokes, thermal, particles, phases, time, fname::String)
+    !isdir(dst) && mkpath(dst) # create folder in case it does not exist
+    jldsave(
+        fname;
+        stokes=Array(stokes),
+        thermal=Array(thermal),
+        particles=Array(particles),
```

Comment:
I think it's best to keep the checkpointing of the particles within JR; I will add a jld2 saving function inside JP. I prefer not to mix the pkgs as long as we can, so that we keep things contained and more simple.